src/System.Device.Gpio/System/Device/Gpio/Drivers/LibgpiodCapiDriver.cs

```diff
@@ -0,0 +1,275 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Runtime.InteropServices;
+
+namespace System.Device.Gpio.Drivers
+{
+    public class LibgpiodCapiDriver : GpioDriver
+    {
+        private const string BasePath = "/dev/";
+
+        public string GpioChip { get; set; } = "gpiochip0";
+
+        private SafeChipHandle _chip;
+
+        private Dictionary<int, SafeLineHandle> _pinNumberToSafeLineHandle;
+
+        private timespec _defaultTimeOut = new timespec(IntPtr.Zero, 1_000_000); // one millisecond
+
+        private Dictionary<int, LibgpiodDriverEventHandler> _pinNumberToEventHandler = new Dictionary<int, LibgpiodDriverEventHandler>();
+
+        protected internal override int PinCount
+        {
+            get
+            {
+                return Interop.gpiod_chip_num_lines(_chip);
+            }
+        }
+
+        public LibgpiodCapiDriver()
+        {
+            OpenChip();
+            _pinNumberToSafeLineHandle = new Dictionary<int, SafeLineHandle>(PinCount);
+        }
+
+        public LibgpiodCapiDriver(string chip)
```

Comment:
what is the scenario for this constructor? People will normally be initializing the controller which should automatically pick a driver. That controller won't know which chip to select yet, so I would instead have this driver not assume any chip, and look for the different pins on all available chips instead.